tests/components/decora_wifi/test_light.py

```diff
@@ -0,0 +1,222 @@
+    """Test light state changes."""
+    # Arrange Fake API State
+    FakeDecoraWiFiIotSwitch.reset_counter()
+    FakeDecoraWiFiSession.clear_accounts()
+    FakeDecoraWiFiSession.add_account(
+        FakeDecoraWiFiAccount(TEST_USERNAME, TEST_PASSWORD, switch_models=["D26HD"])
+    )
+    # Conduct Tests
+    with patch(
+        "homeassistant.components.decora_wifi.common.DecoraWiFiSession",
+        side_effect=FakeDecoraWiFiSession,
+    ), patch(
+        "homeassistant.components.decora_wifi.common.Residence",
+        side_effect=FakeDecoraWiFiResidence,
+    ), patch(
+        "homeassistant.components.decora_wifi.common.ResidentialAccount",
+        side_effect=FakeDecoraWiFiResidentialAccount,
+    ):
+        # Set up the integration
+        await setup_platform(hass)
+
+        # Assert the entity state
+        light_1_a = hass.states.get("light.fake_switch_1")
+        assert light_1_a is not None
+        assert light_1_a.state == "on"
+        assert light_1_a.attributes["brightness"] == 255
+
+        # Send a turn off command
+        await hass.services.async_call(
+            "light", "turn_off", {"entity_id": "light.fake_switch_1"}, blocking=True
+        )
+
+        # Verify the state transition
+        light_1_b = hass.states.get("light.fake_switch_1")
+        assert light_1_b.state == "off"
+
+        # Send a turn on command with brightness and transition speed
+        await hass.services.async_call(
+            "light",
+            "turn_on",
+            {
+                "entity_id": "light.fake_switch_1",
+                ATTR_BRIGHTNESS: 127,
+                ATTR_TRANSITION: 5,
+            },
+            blocking=True,
+        )
+
+        # Verify the state transition
+        light_1_c = hass.states.get("light.fake_switch_1")
+        assert light_1_c is not None
+        assert light_1_c.state == "on"
+        assert light_1_c.attributes[ATTR_BRIGHTNESS] == int(
+            int(127 * 100 / 255) * 255 / 100
+        )
+
+
+async def test_comm_failure(hass: HomeAssistant):
+    """Test behavior against simulated API comm failure."""
+    # Arrange Fake API State
+    FakeDecoraWiFiIotSwitch.reset_counter()
+    FakeDecoraWiFiSession.clear_accounts()
+    FakeDecoraWiFiSession.add_account(
+        FakeDecoraWiFiAccount(TEST_USERNAME, TEST_PASSWORD, switch_models=["D26HD"])
+    )
+    # Conduct Tests
+    with patch(
+        "homeassistant.components.decora_wifi.common.DecoraWiFiSession",
+        side_effect=FakeDecoraWiFiSession,
+    ), patch(
+        "homeassistant.components.decora_wifi.common.Residence",
+        side_effect=FakeDecoraWiFiResidence,
+    ), patch(
+        "homeassistant.components.decora_wifi.common.ResidentialAccount",
+        side_effect=FakeDecoraWiFiResidentialAccount,
+    ):
+        # Set up the integration
+        config_entry = await setup_platform(hass)
+        session = hass.data[DOMAIN][config_entry.entry_id]._session
```

Comment:
Please don't access `hass.data[DOMAIN]` in tests

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Reply:
patch `homeassistant.components.decora_wifi.common.DecoraWiFiSession` to return the mock session object instead and manipulate that.